src/NodaTime/Duration.cs

```diff
@@ -172,6 +172,41 @@ private Duration(long units, string paramName, long minValue, long maxValue, lon
             }
         }
 
+        // Note: unitsPerDay is always actually an integer, but we always want to perform floating point arithmetic
+        // with it.
+        private Duration(double units, string paramName, double unitsPerDay, long nanosPerUnit)
+        {
+            // Note: we don't check the original double values other than for NaN and infinity.
+            // Instead, we do the relevant conversion and then check that we're in bounds.
+            // That way we won't get invalid values due to weird rounding issues.
```

Comment:
I think I get what you're saying here (easier and more reliable to post-verify than attempt to work out exactly when the value will exceed some computed limit), but is there some way to write this comment that doesn't make it sound like we don't know how rounding works?

Reply:
Will have a go.